source/Various/File2.lagda

```diff
@@ -0,0 +1,119 @@
+Alice Laroche, 26th September 2023
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K --exact-split #-}
+
+open import MLTT.Spartan
+open import UF.Univalence
+
+module Various.File2
+        (ua : Univalence)
+        (𝓤 : Universe)
+       where
+
+open import MLTT.NaturalNumbers
+open import UF.Base
+open import UF.DiscreteAndSeparated
+open import UF.Embeddings
+open import UF.Sets
+open import UF.Subsingletons
+open import W.Type
+open import Iterative.Multisets 𝓤
+open import Iterative.Multisets-Addendum ua 𝓤
+open import Iterative.Sets ua 𝓤
+open import Iterative.Sets-Addendum ua 𝓤
+open import Iterative.Ordinals ua 𝓤
+open import Various.File1 ua 𝓤
+
+φ≠ssupφ : (m : 𝕄) (x : 𝕄-root m) → 𝕄-forest m x ≠ m
+φ≠ssupφ (ssup X φ) x eq = φ≠ssupφ (φ x) y eq'
+ where
+  y : 𝕄-root (φ x)
+  y = transport⁻¹ 𝕄-root eq x
+
+  eq' : 𝕄-forest (φ x) y ＝ φ x
+  eq' = transportd 𝕄-root (λ -₁ -₂ → 𝕄-forest -₁ -₂ ＝ -₁) x (eq ⁻¹) eq
+
+succᴹ : 𝕄 → 𝕄
+succᴹ m = ssup (𝕄-root m + 𝟙 {𝓤}) (cases (𝕄-forest m) (λ ⋆ → m))
+
+𝕟ᴹ : ℕ → 𝕄
+𝕟ᴹ 0 = 𝟘ᴹ
+𝕟ᴹ (succ n) = succᴹ (𝕟ᴹ n)
+
+succᴹ-preserve-iset : (m : 𝕄)
+                    → is-iterative-set m
+                    → is-iterative-set (succᴹ m)
+succᴹ-preserve-iset m is-iset = III , IV
+ where
+  I : is-h-isolated m
+  I = isets-are-h-isolated m is-iset
+
+  II : is-embedding (λ _ → m)
+  II = global-point-is-embedding (λ _ → m) I
+
+  III : is-embedding (𝕄-forest (succᴹ m))
+  III = disjoint-cases-embedding _ _ (𝕄-forest-is-embedding m is-iset) II (λ x ⋆ → φ≠ssupφ m x)
```

Comment:
I would break it like this
```
III = disjoint-cases-embedding _ _ 
        (𝕄-forest-is-embedding m is-iset) 
        II 
        (λ x ⋆ → φ≠ssupφ m x)
```
or like this
```
III = disjoint-cases-embedding _ _ 
        (𝕄-forest-is-embedding m is-iset) II (λ x ⋆ → φ≠ssupφ m x)        
```
I think I prefer the first one, but both are acceptable.